BasicEngine.Tests.ps1: Add -noprofile to pwsh invocation

MatejKafka opened the pull request:
# PR Summary

Fixes a test that's invoking a new pwsh instance, but does not pass `-noprofile`. I'm printing a custom banner in my profile, which fails the test, as it adds unexpected output.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
  - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge. If this PR is a work in progress, please open this as a [Draft Pull Request and mark it as Ready to Review when it is ready to merge](https://docs.github.com/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests#draft-pull-requests).
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
  - [x] None
  - **OR**
  - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.5/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
    - [ ] Experimental feature name(s): <!-- Experimental feature name(s) here -->
- **User-facing changes**
  - [x] Not Applicable
  - **OR**
  - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - [ ] Issue filed: <!-- Number/link of that issue here -->
- **Testing - New and feature**
  - [x] N/A or can only be tested interactively
  - **OR**
  - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)

MatejKafka committed:
BasicEngine.Tests.ps1: Add -noprofile to pwsh invocation

iSazonov commented:
@MatejKafka I found the same in Job.Tests.ps1. Could you please review all test files for the issue and add fixes?

MatejKafka committed:
Fix more missing -noprofile tests, clean up ConsoleHost.Tests.ps1

MatejKafka commented:
I did not go through all tests, but grepped for a few common patterns and fixed what I found and seemed fishy. Also cleaned up some of the test files a bit (added util includes and strict mode / error action setup) and fixed one minor bug.

muffy86 commented:
This pull request appears inactive. If there are plans to continue work, please update the PR within 14 days; otherwise it may be closed.
